upstream-sha256: 6a2082a6f94773971e005f6521063b6905dafd1bddcda76c5f12a3caf41e1fd0
actual-sha256: 187f95c005e67ad04b1cfe3450255e3ffc4cb075a42390726bf967485d064fff
--- upstream.expected
+++ tolk.actual
@@ -29,10 +29,10 @@
         float val13 = (alu8?*(data1_3000+(alu4+702)):0.0f);
         float val14 = (alu7?*(data1_3000+(alu4+705)):0.0f);
         float val15 = (alu6?*(data1_3000+(alu4+708)):0.0f);
-        *(buf0+0) = ((*(buf0+0))+val7+val6+val5+val4);
-        *(buf0+1) = ((*(buf0+1))+val3+val2+val1+val0);
-        *(buf0+2) = ((*(buf0+2))+val12+val13+val14+val15);
-        *(buf0+3) = ((*(buf0+3))+val8+val9+val10+val11);
+        *(buf0+0) = ((*(buf0+0))+(val7+val6+val5+val4));
+        *(buf0+1) = ((*(buf0+1))+(val3+val2+val1+val0));
+        *(buf0+2) = ((*(buf0+2))+(val12+val13+val14+val15));
+        *(buf0+3) = ((*(buf0+3))+(val8+val9+val10+val11));
       }
       int alu15 = ((Lidx1<<10)+Lidx3);
       *(data0_3072+(alu15+256)) = (*(buf0+1));
@@ -51,9 +51,10 @@
     float val1 = (*(data1_3072+(alu0+255)));
     float val2 = (*(data1_3072+(alu0+511)));
     float val3 = (*(data1_3072+(alu0+767)));
-    float alu1 = (val1+val2);
-    float alu2 = (alu1+val3);
-    *((float4*)((data0_12+(Lidx1<<2)))) = (float4){val1,alu1,alu2,(alu2+val0)};
+    float alu1 = (0.0f+val1);
+    float alu2 = (alu1+val2);
+    float alu3 = (alu2+val3);
+    *((float4*)((data0_12+(Lidx1<<2)))) = (float4){alu1,alu2,alu3,(alu3+val0)};
   }
 }
 ---
